sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/PreDefinedTagsOperations.cs

```diff
@@ -0,0 +1,202 @@
+            scope.Start();
+            try
+            {
+                return RestClient.CreateOrUpdateValue(tagName, tagValue, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> This operation allows deleting a name from the list of predefined tag names for the given subscription. The name being deleted must not be in use as a tag name for any resource. All predefined values for the given name must have already been deleted. </summary>
+        /// <param name="tagName"> The name of the tag. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response> DeleteAsync(string tagName, CancellationToken cancellationToken = default)
+        {
+            using var scope = Diagnostics.CreateScope("PreDefinedTagsOperations.Delete");
+            scope.Start();
+            try
+            {
+                var operation = await StartDeleteAsync(tagName, cancellationToken).ConfigureAwait(false);
+                return await operation.WaitForCompletionResponseAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> This operation allows deleting a name from the list of predefined tag names for the given subscription. The name being deleted must not be in use as a tag name for any resource. All predefined values for the given name must have already been deleted. </summary>
+        /// <param name="tagName"> The name of the tag. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual Response Delete(string tagName, CancellationToken cancellationToken = default)
+        {
+            using var scope = Diagnostics.CreateScope("PreDefinedTagsOperations.Delete");
+            scope.Start();
+            try
+            {
+                var operation = StartDelete(tagName, cancellationToken);
+                return operation.WaitForCompletion(cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> This operation allows deleting a name from the list of predefined tag names for the given subscription. The name being deleted must not be in use as a tag name for any resource. All predefined values for the given name must have already been deleted. </summary>
+        /// <param name="tagName"> The name of the tag. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Operation> StartDeleteAsync(string tagName, CancellationToken cancellationToken = default)
```

Comment:
method should return the derived operation type
```suggestion
        public virtual async Task<PreDefinedTagDeleteOperation> StartDeleteAsync(string tagName, CancellationToken cancellationToken = default)
```

Reply:
Done